src/test/java/uk/gov/companieshouse/api/accounts/controller/DirectorsApprovalControllerTest.java

```diff
@@ -0,0 +1,326 @@
+                directorsApproval);
+        doReturn(successFindResponse).when(directorsApprovalService).find("123456", request);
+
+        ResponseEntity responseEntity = ResponseEntity.status(HttpStatus.OK).body(directorsApproval);
+        when(apiResponseMapper.mapGetResponse(directorsApproval,
+                request)).thenReturn(responseEntity);
+
+        ResponseEntity response = directorsApprovalController.get("123456", request);
+
+        assertNotNull(response);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(directorsApproval, response.getBody());
+    }
+
+    @Test
+    @DisplayName("create directors approval resource throws mongo exception")
+    void createDirectorsApprovalThrowsMongoException() throws DataException {
+
+        when(request.getAttribute(AttributeName.TRANSACTION.getValue())).thenReturn(transaction);
+
+        when(directorsApprovalService.find(COMPANY_ACCOUNTS_ID, request))
+                .thenThrow(new DataException(""));
+
+        when(apiResponseMapper.getErrorResponse())
+                .thenReturn(createResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR, null));
+
+        ResponseEntity response =
+                directorsApprovalController.get(COMPANY_ACCOUNTS_ID, request);
+
+        assertDirectorsApprovalControllerResponse(response, HttpStatus.INTERNAL_SERVER_ERROR, false);
+        verify(directorsApprovalService, times(1))
+                .find(COMPANY_ACCOUNTS_ID, request);
+        verify(apiResponseMapper, times(1)).getErrorResponse();
+    }
+/*
+    @Test
+    @DisplayName("Create directors approval resource - has binding errors")
+    void createDirectorsApprovalResourceBindingErrors() {
+
+        when(bindingResult.hasErrors()).thenReturn(true);
+        when(errorMapper.mapBindingResultErrorsToErrorModel(bindingResult)).thenReturn(new Errors());
+
+        ResponseEntity responseEntity =
+                directorsApprovalController.create(directorsApproval, bindingResult,
+                        COMPANY_ACCOUNTS_ID, request);
+
+        assertNotNull(responseEntity);
+        assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
+    }
+*/
+    @Test
+    @DisplayName("Update directors approval resource - no directors approval link")
+    void updateDirectorsApprovalResourceNoDirectorsApprovalLink() {
+
+        when(request.getAttribute(anyString())).thenReturn(directorsReport).thenReturn(transaction);
+        when(directorsReport.getLinks()).thenReturn(directorsReportLink);
+        when(directorsReportLink.get(DirectorsReportLinkType.APPROVAL.getLink())).thenReturn(null);
+
+        ResponseEntity responseEntity =
+                directorsApprovalController.update(directorsApproval,
+                        COMPANY_ACCOUNTS_ID, request);
+
+        assertNotNull(responseEntity);
+        assertEquals(HttpStatus.NOT_FOUND, responseEntity.getStatusCode());
+        assertNull(responseEntity.getBody());
+    }
+/*
```

Comment:
Should this be commented?

Reply:
Removed the commented block as it is not required at the moment